### DISCUSSION TITLE
why did the String Loader interface got removed?

### DISCUSSION THREAD

**Discussion by AhmadAlassad** [B3nedikt/restring#148] [Q&A] [answered]
Hello, 
thanks for the effort.
may I ask why did you remove the String Loader interface and isn't it is purpose to load the strings in background thread?
if I load the strings directly from the main thread could that affect the performance?
Regards.
Ahmad

**Comment by B3nedikt** [comment, marked as answer]
Yes, you should of course always make network requests on a background thread, e.g. using kotlin coroutines. But the exact way depends very much on your use case, which is why I removed the interface. So you should just load the strings first, and then just supply them to restring when you are finished loading. After this you need to restart the activity, or use my Reword library as described in the Readme.
